plugin/advisor/catalog/walk_through.go

```diff
@@ -0,0 +1,420 @@
+		keyList, err := t.validateAndGetKeyStringList(constraint.Keys, true)
+		if err != nil {
+			return err
+		}
+		if err := t.createPrimaryKey(keyList, getIndexType(constraint.Option)); err != nil {
+			return err
+		}
+	case tidbast.ConstraintKey, tidbast.ConstraintIndex:
+		keyList, err := t.validateAndGetKeyStringList(constraint.Keys, false)
+		if err != nil {
+			return err
+		}
+		if err := t.createIndex(constraint.Name, keyList, false, getIndexType(constraint.Option)); err != nil {
+			return err
+		}
+	case tidbast.ConstraintUniq, tidbast.ConstraintUniqKey, tidbast.ConstraintUniqIndex:
+		keyList, err := t.validateAndGetKeyStringList(constraint.Keys, false)
+		if err != nil {
+			return err
+		}
+		if err := t.createIndex(constraint.Name, keyList, true, getIndexType(constraint.Option)); err != nil {
+			return err
+		}
+	case tidbast.ConstraintForeignKey:
+		// we can not deal with FOREIGN KEY constraints
+	case tidbast.ConstraintFulltext:
+		keyList, err := t.validateAndGetKeyStringList(constraint.Keys, false)
+		if err != nil {
+			return err
+		}
+		if err := t.createIndex(constraint.Name, keyList, false, FullTextName); err != nil {
+			return err
+		}
+	case tidbast.ConstraintCheck:
+		// we can not deal with CHECK constraints
+	}
+
+	return nil
+}
+
+func (t *tableState) validateAndGetKeyStringList(keyList []*tidbast.IndexPartSpecification, primary bool) ([]string, error) {
+	var res []string
+	for _, key := range keyList {
+		if key.Expr != nil {
+			str, err := restoreNode(key, format.DefaultRestoreFlags)
+			if err != nil {
+				return nil, err
+			}
+			res = append(res, str)
+		} else {
+			columnName := key.Column.Name.O
+			column, exists := t.columnSet[columnName]
+			if !exists {
+				return nil, &WalkThroughError{
+					Type:    ErrorTypeColumnNotExists,
+					Content: fmt.Sprintf("Column `%s` in table `%s` not exists", columnName, t.name),
+				}
+			}
+			if primary {
+				column.nullable = false
+			}
+			res = append(res, columnName)
+		}
+	}
+	return res, nil
+}
+
+func (t *tableState) createColumn(column *tidbast.ColumnDef, position int) error {
+	if _, exists := t.columnSet[column.Name.Name.O]; exists {
+		return &WalkThroughError{
+			Type:    ErrorTypeColumnExists,
+			Content: fmt.Sprintf("Column `%s` exists in table `%s`", column.Name.Name.O, t.name),
+		}
+	}
+
+	col := &columnState{
+		name:         column.Name.Name.O,
+		position:     position,
+		columnType:   column.Tp.CompactStr(),
+		characterSet: column.Tp.GetCharset(),
+		collation:    column.Tp.GetCollate(),
+		nullable:     true,
+	}
+
+	for _, option := range column.Options {
+		switch option.Tp {
+		case tidbast.ColumnOptionPrimaryKey:
+			col.nullable = false
+			if err := t.createPrimaryKey([]string{col.name}, model.IndexTypeBtree.String()); err != nil {
+				return err
+			}
+		case tidbast.ColumnOptionNotNull:
+			col.nullable = false
+		case tidbast.ColumnOptionAutoIncrement:
+			// we can not deal with AUTO-INCREMENT
```

Comment:
Is this a TODO and should return NOT Implemented? same below

Reply:
The same as CONSTRAINT